### PR TITLE
Improve losses tests, add `TestSSIM3d`, and `BaseTester.gradcheck`

### LINKED ISSUE #2136: Improve loss tests
- add BaseTester to tests merged on #2130
  - Check if we can have a better testing for the loss calculus itself, instead of just assert against 0 -- can check this for all losses
- improve the tests merged on #2109
  - decrease input size, use BaseTester, and other requests made in the PR

### PR DESCRIPTION
**Pull request by johnnv1**
#### Changes
Fixes #2136 

#### Type of change
- [x] 📚  Documentation Update
  - add to docs the losses: `Welsch`, `Cauchy`, `GemanMcclureLoss` and `Charbonnier`
- [x] 🧪 Tests Cases
   - Improve some losses tests
   - add missing test for `kornia.metrics.SSIM3D`
- [x] 🔬 New feature (non-breaking change which adds functionality)
  -  add gradcheck method to `BaseTester`

### CODE REVIEW

File: test/losses/test_ssim.py [hunk 1]
```diff
@@ -171,19 +181,30 @@ def test_module(self, device, dtype):
...
         # input data
-        window_size = 3
-        img1 = torch.rand(1, 1, 5, 4, 3, device=device, dtype=dtype)
-        img2 = torch.rand(1, 1, 5, 4, 3, device=device, dtype=dtype)
+        img = torch.rand(1, 1, 5, 4, 3, device=device)
 
         # evaluate function gradient
-        img1 = utils.tensor_to_gradcheck_var(img1)  # to var
-        img2 = utils.tensor_to_gradcheck_var(img2)  # to var
+        img = utils.tensor_to_gradcheck_var(img)  # to var
```
Comment (edgarriba) [inline test/losses/test_ssim.py:191]: As future feature,  maybe tensor to gradcheck should be also in the base tester

**Review by edgarriba — APPROVED** [review]
Minor comments and ideas

File: test/losses/test_ssim.py [hunk 2]
```diff
@@ -123,33 +123,43 @@ def test_jit(self, device, dtype):
...
+        # Check loss computation
+        img1 = torch.ones(shape, device=device, dtype=dtype)
+        img2 = torch.zeros(shape, device=device, dtype=dtype)
 
-        tol_val: float = utils._get_precision_by_name(device, 'xla', 1e-1, 1e-4)
-        assert_close(loss.item(), 0.0, rtol=tol_val, atol=tol_val)
+        actual = kornia.losses.ssim3d_loss(img1, img2, window_size, reduction=reduction_type)
+
+        if reduction_type == 'mean':
+            expected = torch.tensor(0.9999, device=device, dtype=dtype)
```
[the remark this answers is not in the document]
Comment (johnnv1) [reply test/losses/test_ssim.py:158]: I don't understand what you mean

File: test/losses/test_ssim.py (hunk 1 above)
Comment (johnnv1) [reply test/losses/test_ssim.py:191]: yep, and maybe we can apply to all tensors on the inputs of `self.gradcheck`

Comment (edgarriba) [reply test/losses/test_ssim.py:191]: Agree

File: test/losses/test_ssim.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (edgarriba) [reply test/losses/test_ssim.py:158]: The assert close should work if you set to 1.0

[the remark this answers is not in the document]
Comment (johnnv1) [reply test/losses/test_ssim.py:158]: ah, for the mean case yes, for the accumulative/sum will not work -- I tried to make it closer to the real thing
